cpp/src/copying/contiguous_split.cu

```diff
@@ -243,36 +243,22 @@ struct column_copy_functor {
          return;
       }
 
-      // custom copy kernel (which should probably just be an in-place copy() function in cudf.
+      // custom copy kernel (which could probably just be an in-place copy() function in cudf).
       cudf::size_type num_els = cudf::util::round_up_safe(in.size(), cudf::experimental::detail::warp_size);
       constexpr int block_size = 256;
       cudf::experimental::detail::grid_1d grid{num_els, block_size, 1};
       
-      // so there's a significant performance issue that comes up. our incoming column_view objects
-      // are the result of a slice.  because of this, they have an UNKNOWN_NULL_COUNT.  because of that,
-      // calling column_device_view::create() will cause a recompute of the count, which ends up being
-      // extremely slow because a.) the typical use case here will involve huge numbers of calls and
-      // b.) the count recompute involves tons of device allocs and memcopies.
-      //
-      // so to get around this, I am manually constructing a fake-ish view here where the null
-      // count is arbitrarily bashed to 0.
-      //            
-      // Remove this hack once rapidsai/cudf#3600 is fixed.
-      column_view   in_wrapped{in.type(), in.size(), in.head<T>(), 
-                               in.null_mask(), in.null_mask() == nullptr ? UNKNOWN_NULL_COUNT : 0,
-                               in.offset() };
-      mutable_column_view  mcv{in.type(), in.size(), data, 
-                               validity, validity == nullptr ? UNKNOWN_NULL_COUNT : 0 };
+      // output copied column
+      mutable_column_view  mcv{in.type(), in.size(), data, validity, in.null_count()};
       if(in.nullable()){
```

Comment:
consider  `in.has_nulls()` instead of `in.nullable()`

Reply:
Yep, that makes sense.  Especially in this case, where we're now guaranteed to have a valid null count.  I'll change.